app/src/main/java/ca/marshallasch/veil/PostListAdapter.java

```diff
@@ -190,6 +196,55 @@ public void sort(SortOption option) {
      */
     @Override
     public int getItemCount() {
-        return posts.size();
+        return postsFiltered.size();
+    }
+
+
+    @Override
+    public Filter getFilter()
+    {
+
+        /**
+         * This class is the tag filter. The CharSequence must be a list of tags that are
+         * denominated with a ':' character.
+         */
+        return new Filter() {
+            @Override
+            protected FilterResults performFiltering(CharSequence charSequence)
+            {
+
+                List<DhtProto.Post> filteredList = new ArrayList<>();
+
+                if (charSequence.length() == 0) {
+                    filteredList = posts;
+                } else {
+                    String[] tokens = charSequence.toString().split(":");
+
+                    List<String> tagList = new ArrayList<>();
+
+                    for (String tag: tokens){
+                        tagList.add(tag.trim());
+                    }
+
+                    // make sure the post contains all the tags
```

Comment:
What do you mean by this comment? Should all posts have all the tags?

Reply:
It  means that it checks that the post has all the tags it is filtering by.

Reply:
alright sounds good just make the comment more clear thank!